utils/common/src/main/kotlin/Utils.kt

```diff
@@ -128,3 +128,35 @@ fun <R> temporaryProperties(vararg properties: Pair<String, String?>, block: ()
         }
     }
 }
+
+/**
+ * Return a comma-separated string of values or ranges that represent the [current object].
+ */
+fun List<Pair<Int, Int>>.prettyPrintValues(): String {
+    return joinToString { (startValue, endValue) ->
+        if (startValue == endValue) startValue.toString() else "$startValue-$endValue"
+    }
+}
+
+/**
+ * Collapse consecutive [values] to a list of pairs that each denote a range. A single value is represented as a
+ * range whose first and last elements are equal.
+ */
+fun collapseValues(values: Collection<Int>): List<Pair<Int, Int>> {
```

Comment:
I'd prefer a function name that better explains *what to* values are collapsed. Maybe "createRanges", "collapseToRanges", "extractRanges", or something like that.